### PR TITLE
Edits to follow up contribution guidelines work

### LINKED ISSUE #897: Contribution guidelines needs rework or should be dropped
We have the following "contributing" resources that overlap (at least in part) in their content:

- https://opentelemetry.io/docs/contribution-guidelines/
- https://github.com/open-telemetry/opentelemetry.io/blob/main/CONTRIBUTING.md
- https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md

We should consolidate these resources to avoid overlap. One idea is to drop the website page, and simply refer interested readers to one of the `CONTRIBUTING.md` pages.

Related to #770.

### LINKED ISSUE #3767: Updates to Contributing info
This is a tracker issue, following up on:

- #3651

### Tasks

- [ ] Wrap up #897 by consolidating any remaining content of https://opentelemetry.io/docs/contribution-guidelines/ into the main Contributing page
- [ ] Fix aliases to avoid 404s
- [ ] ...

/cc @theletterf

### PR DESCRIPTION
**Pull request by theletterf**
This PR consolidates contribution docs further and erases the hidden `contribution-guidelines` doc, which was redundant.

- Fixes #3767 
- Fixes #897 

**Preview**: https://deploy-preview-3846--opentelemetry.netlify.app/docs/contributing/

**Redirect test**: https://deploy-preview-3846--opentelemetry.netlify.app/docs/contribution-guidelines

### CODE REVIEW

**Comment by theletterf** [comment]
@svrnm @chalin Did the mirrored documentation guidelines change / were removed? I don't think we've them in prod: https://github.com/open-telemetry/opentelemetry.io/commit/7c1dcce3344b33cf1f1877a3a08ae3e8f4ef3d0e

Can't trace when/where they were removed. I take they no longer apply, as we use a more generic language in our new contribution guide. As they were still linked from contribution-guidelines I was wondering.

**Review by svrnm — APPROVED** [review]
lgtm, but I'd like to also get @chalin's approval

File: .htmltest.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ TestFilesConcurrently: true
 IgnoreDirs:
   - ^blog/(\d+/)?page/\d+
 IgnoreInternalURLs: # list of paths
+  - /docs/contribution-guidelines/
```
Comment (chalin) [inline .htmltest.yml:11]: Ah, if you modify the community list shortcode, then you don't need to ignore the path anymore:

```suggestion
```

**Review by chalin — APPROVED** [review]
One suggested tweak, otherwise LGTM overall :)

**Review by chalin — CHANGES_REQUESTED** [review]
Oh! We need to add an alias for the page that was removed.

**Comment by chalin** [comment]
> @svrnm @chalin Did the mirrored documentation guidelines change / were removed? I don't think we've them in prod: [7c1dcce](https://github.com/open-telemetry/opentelemetry.io/commit/7c1dcce3344b33cf1f1877a3a08ae3e8f4ef3d0e)

Those guidelines no longer apply.

For context, see:

- #833
- #730

**Comment by chalin** [comment]
> Also, do we want to promote the newly created https://github.com/open-telemetry/opentelemetry.io/discussions board? I'd wait a bit before doing so, perhaps...

I'm ok with waiting a bit too.

**Comment by theletterf** [comment]
@chalin Changes applied, thanks!

The alias is already here I think? https://github.com/open-telemetry/opentelemetry.io/pull/3846/files#diff-a3d923889a7028820da3c39a743bd702eb33e7333b64009b1384d444c7a1561aL4

**Comment by chalin** [comment]
> @chalin Changes applied, thanks!

👍 

> The alias is already here I think? https://github.com/open-telemetry/opentelemetry.io/pull/3846/files#diff-a3d923889a7028820da3c39a743bd702eb33e7333b64009b1384d444c7a1561aL4

Yes! 👍🏻

**Review by chalin — APPROVED** [review]
👍🏻

**Pull request merged**